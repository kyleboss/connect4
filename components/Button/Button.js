import styled from "styled-components";
import { TABLET_AND_LARGER_SIZE } from "@/utils/constants";

const StyledButton = styled.button`
  font-size: 1rem;
  padding: 1rem 1.5rem;
  opacity: 0.9;
  transition: opacity 0.3s ease, box-shadow 0.3s ease;
  cursor: pointer;
  border-radius: 10px;

  &:hover {
    opacity: 1;
    box-shadow: 0 0 10px rgba(255, 255, 255, 0.5);
  }

  @media (min-width: ${TABLET_AND_LARGER_SIZE}) {
    font-size: 2rem;
    padding: 1.5rem 2rem;
  }
`;

const Button = ({ children, onClick }) => {
  return <StyledButton onClick={onClick}>{children}</StyledButton>;
};

export default Button;
