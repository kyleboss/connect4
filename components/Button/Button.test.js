import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

import Button from "./Button";

describe("Button", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Button>Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls the onClick handler when clicked", () => {
    const onClickMock = jest.fn();
    render(<Button onClick={onClickMock}>Click me</Button>);
    fireEvent.click(screen.getByText("Click me"));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it("displays the correct children", () => {
    render(<Button>Test Button</Button>);
    expect(screen.getByText("Test Button")).toBeInTheDocument();
  });
});
