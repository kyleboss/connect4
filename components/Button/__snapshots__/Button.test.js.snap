// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-5e2d1029-0 NWVxU"
  >
    Click me
  </button>
</DocumentFragment>
`;
